backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/bom/asiSupportOnProductBOM.feature

```diff
@@ -0,0 +1,448 @@
+    And verify that bom was created for product
+      | PP_Product_BOM_ID.Identifier | M_Product_ID.Identifier | PP_Product_BOMVersions_ID.Identifier | ProductValue | UomCode | ValidFrom                   | OPT.M_AttributeSetInstance_ID.Identifier |
+      | bom_1                        | product_S1              | bv_1                                 | product_S1   | PCE     | 2022-01-03T11:09:47.000000Z | bomAttributeSetInstance                  |
+    And validate M_AttributeInstance:
+      | M_AttributeSetInstance_ID.Identifier | Value    |
+      | bomAttributeSetInstance              | Bio,GMAA |
+
+    And verify that bomLine was created for bom
+      | OPT.PP_Product_BOMLine_ID.Identifier | PP_Product_BOM_ID.Identifier | M_Product_ID.Identifier | Qty | UomCode | IsQtyPercentage | Scrap | Line | OPT.M_AttributeSetInstance_ID.Identifier |
+      | bomLine_1                            | bom_1                        | component_S1            | 5.0 | PCE     | false           | 0     | 1    | bomLineAttributeSetInstance              |
+    And validate M_AttributeInstance:
+      | M_AttributeSetInstance_ID.Identifier | Value |
+      | bomLineAttributeSetInstance          | Bio   |
+
+    And verify BOM for M_Product:
+      | M_Product_ID.Identifier |
+      | product_S1              |
+
+    And metasfresh contains M_AttributeSetInstance with identifier "ppProductPlanningAttributeSetInstance":
+  """
+  {
+    "attributeInstances":[
+      {
+        "attributeCode":"1000002",
+          "valueStr":"Bio"
+      }
+    ]
+  }
+  """
+    And metasfresh contains PP_Product_Plannings
+      | Identifier      | M_Product_ID.Identifier | PP_Product_BOMVersions_ID.Identifier | IsCreatePlan | OPT.M_AttributeSetInstance_ID.Identifier | IsAttributeDependant |
+      | pp_finishedGood | product_S1              | bv_1                                 | false        | ppProductPlanningAttributeSetInstance    | true                 |
+
+    And metasfresh contains M_AttributeSetInstance with identifier "orderLineAttributeSetInstance":
+  """
+  {
+    "attributeInstances":[
+      {
+        "attributeCode":"1000015",
+          "valueStr":"03"
+      }
+    ]
+  }
+  """
+    And metasfresh contains C_Orders:
+      | Identifier | IsSOTrx | C_BPartner_ID.Identifier | DateOrdered | M_PricingSystem_ID.Identifier |
+      | order_SO   | Y       | customer_SO              | 2022-01-03  | ps_SO                         |
+    And metasfresh contains C_OrderLines:
+      | Identifier   | C_Order_ID.Identifier | M_Product_ID.Identifier | QtyEntered | OPT.M_AttributeSetInstance_ID.Identifier |
+      | orderLine_SO | order_SO              | product_S1              | 5          | orderLineAttributeSetInstance            |
+    And the order identified by order_SO is completed
+
+    And after not more than 30s, no PP_Order_Candidates are found:
+      | C_OrderLine_ID.Identifier |
```

Comment:
assume that the PP_Order_Candidates are created after 5secs. wouldn't this test still succeed? because after 500ms there is no PP_Order_Candidate?
so it seems to me that can't work

What would work though AFAIU is, to wait for de.metas.material.planning.pporder.PPOrderCandidateDemandMatcher#matches to write its message into an I_AD_EventLog_Entry record.

Reply:
> assume that the PP_Order_Candidates are created after 5secs. wouldn't this test still succeed? because after 500ms there is no PP_Order_Candidate?

not really, the `tryAndWait` function will keep retrying until `loadPPOrderCandidates` returns true (i.e. finds the PP_Order_Candidate) or the `30s` coming from the feature file are gone.. so it will retry for 30s until it says there is no ppOrderCand created (which is not ideal as that's a lot of waiting time) so I'm happy to go with ur approach in that follow-up 👍